src/temporal_logic/rules.rs

```diff
@@ -1126,6 +1147,38 @@ macro_rules! strengthen_next_n_internal {
 pub use strengthen_next_n;
 pub use strengthen_next_n_internal;
 
+/// Show that an spec entails the invariant by a group of action/state predicates which are also invariants entailed by spec.
+/// pre:
+///     spec |= []next
+///     forall |s, s_prime| next(s, s_prime) ==> inv(s, s_prime)
+///     spec |= []p1
+///     spec |= []p2
+///         ...
+///     spec |= []pn
+///     next == p1 /\ p2 /\ ... /\ pn
+/// post:
+///     spec |= []inv
+/// 
+/// Usage: invariant_action_n!(spec, next, inv, p1, p2, ..., pn)
+#[macro_export]
+macro_rules! invariant_action_n {
```

Comment:
I was about to say that we should make the macros consistent with `strengthen_next_n` on the order of arguments, but I also realized that there is no need to differentiate `next` and other `p` in `strengthen_next_n`. I think putting `spec` before `next` (actually the strengthened next) and treating all others as `px` is the right way to go. I will refactor `strengthen_next_n` in this way later.